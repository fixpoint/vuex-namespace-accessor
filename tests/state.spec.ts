import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

import counter, { CounterState } from './counter';
import { createAccessor } from '..';

const store = new Vuex.Store({
  modules: {
    counter,
  },
});

describe('createAccessor', () => {
  const a = createAccessor<CounterState>(store, 'counter');

  it('returns an accessor which state is linked to the module', () => {
    expect(a.state.count).toEqual(0);
    expect(a.state).toMatchSnapshot();

    store.commit('counter/inc', {
      amount: 20,
    });
    expect(a.state.count).toEqual(20);
    expect(a.state).toMatchSnapshot();

    store.commit('counter/inc', {
      amount: 10,
    });
    expect(a.state.count).toEqual(30);
    expect(a.state).toMatchSnapshot();
  });
});
