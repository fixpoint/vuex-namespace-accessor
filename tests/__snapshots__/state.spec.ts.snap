// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createAccessor returns an accessor which state is linked to the module 1`] = `
Object {
  "count": 0,
}
`;

exports[`createAccessor returns an accessor which state is linked to the module 2`] = `
Object {
  "count": 20,
}
`;

exports[`createAccessor returns an accessor which state is linked to the module 3`] = `
Object {
  "count": 30,
}
`;
